sysl2/sysl/ints.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type IntegrationParam struct {
+	m            *sysl.Module
+	excludes     StrSet
+	passthroughs StrSet
+	commonEndpts StrSet
+	drawableApps StrSet
+	apps         StrSet
+	deps         map[string]*AppDependency
+}
+
+func MakeIntegrationParam(mod *sysl.Module, excludes, passthroughs, highlights StrSet) *IntegrationParam {
+	return &IntegrationParam{
+		m:            mod,
+		excludes:     excludes,
+		passthroughs: passthroughs,
+		commonEndpts: MakeStrSet(".. * <- *", "*"),
+		drawableApps: highlights,
+		apps:         StrSet{},
+		deps:         map[string]*AppDependency{},
+	}
+}
+
+func (v *IntegrationParam) collectApplicationDependencies(apps StrSet) {
+	for app := range apps {
+		apps := v.m.GetApps()
+		for epname, endpt := range apps[app].GetEndpoints() {
+			v.handleStatement(app, epname, endpt.GetStmt())
+		}
+	}
+}
+
+func (v *IntegrationParam) handleStatement(appname, epname string, stmts []*sysl.Statement) {
+	for _, stmt := range stmts {
+		switch t := stmt.GetStmt().(type) {
+		case *sysl.Statement_Call:
+			targetName := getAppName(t.Call.GetTarget())
+			if added := v.addAppDependency(appname, epname, targetName, t.Call.Endpoint, stmt); added {
+				if t.Call.Endpoint != ".. * <- *" {
+					for epname, endpt := range v.m.GetApps()[targetName].GetEndpoints() {
+						v.handleStatement(targetName, epname, endpt.GetStmt())
```

Comment:
why are you iterating over all endpoints here?

Reply:
we need find all the call, right? I think we should iterate each endpoint to do this.

Reply:
no. i think you should only follow the call's dependency.